blog/breach-to-barrier.md

```diff
@@ -0,0 +1,39 @@
+
+Before we talk about _how_ you migrate to the sandbox, let’s first discuss _why_ you want it.
+
+The sandbox puts a hard cage around all renderer processes, ensuring that no matter what happens inside, code is executed inside a restricted environment. As a concept, it's a lot older than Chromium, and provided as a feature by all major operating systems. Electron's and Chromium's sandbox build on top of these system features. Even if you never display user-generated content, you should consider the possibility that your renderer might get compromised: Scenarios as sophisticated as supply chain attacks and as simple as little bugs can lead to your renderer doing things you didn't fully intend for it to do.
+
+The sandbox makes that scenario a lot less scary: A process inside gets to freely use CPU cycles and memory — that’s it. Processes cannot write to disk or display their own windows. In the case of our `libwep` bug, the sandbox makes sure that an attacker cannot install or run malware. In fact, in the case of the original Pegasus attack on the employee’s iPhone, the attack specifically targeted a non-sandboxed image process to gain access to the phone, first breaking out of the boundaries of the normally sandboxed iMessage. When a CVE like the one in this example is announced, you still have to upgrade your Electron apps to a secure version — but in the meantime, the amount of damage an attacker can do is limited dramatically.
+
+Migrating a vanilla Electron application from `sandbox: false` to `sandbox: true` is an undertaking. I know, because even though I have personally written the first draft of the [Electron Security Guidelines](https://www.electronjs.org/docs/latest/tutorial/security), I have not managed to migrate one of my own apps to use it. That changed this weekend, and I recommend that you change it, too.
+
+![Don’t be scared by the number of line changes, most of it is in `package-lock.json`](/assets/img/breach-to-barrier.png)
+
+There are two things you need to tackle:
+
+1. If you’re using Node.js code in either `preload` scripts or the actual `WebContents`, you need to move all that Node.js interaction to the main process (or, if you are fancy, a utility process). Given how powerful renderers have become, chances are high that the vast majority of your code doesn’t really need refactoring.
+
+   Consult our documentation on [Inter-Process Communication](https://www.electronjs.org/docs/latest/tutorial/ipc). In my case, I moved a lot of code and wrapped it in `ipcRenderer.invoke()` and `ipcMain.handle()`, but the process was straightforward and quickly done.
```

Comment:
I think it would be good to add another paragraph to this section with a warning that IPC can provide ways to bypass the sandbox if you're not careful. You should consider how any API you expose to the renderers via IPC might be abused by a compromised renderer. Maybe say use it sparingly to dovetail with the comment in the first paragraph about how powerful renderers can already do a lot of what you likely need.

Reply:
I'll add something but I think I also don't want to scare people away from using Electron in the first place. It's a bit of a balance but I don't want people to walk away from this blog post thinking that there are hidden bombs everywhere